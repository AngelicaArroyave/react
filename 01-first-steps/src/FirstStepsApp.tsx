import { ItemCounter } from './shopping-cart/ItemCounter';

interface ItemInCart {
    productName: string
    quantity: number
}

const itemsInCart: ItemInCart[] = [
    { productName: 'Nintendo Switch 2', quantity: 5 },
    { productName: 'Pro Controller', quantity: 3 },
    { productName: 'Super Smash', quantity: 4 }
]

export function FirstStepsApp() {
    return (
        <>
            <h1>Shopping Cart</h1>
            
            {/* <ItemCounter name="Nintendo Switch 2" quantity={5} />
            <ItemCounter name="Pro Controller" quantity={3} />
            <ItemCounter name="Super Smash" quantity={1} /> */}

            { itemsInCart.map(({ productName, quantity }) => (
                <ItemCounter key={productName} name={productName} quantity={quantity} />
            )) }
        </>
    )
}