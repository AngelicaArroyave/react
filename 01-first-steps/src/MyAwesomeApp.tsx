import type { CSSProperties } from 'react'

const firstName = 'Angie'
const lastName = 'Arias'

const favoriteGames = ['Elden Ring', 'Smash', 'Metal Gear']

const isActive = true

const address = {
    zipCode: 'ABC-123',
    country: 'Canada'
}

const myStyles: CSSProperties = {
                backgroundColor: 'purple',
                color: 'white',
                borderRadius: isActive ? 8 : 15,
                padding: 10,
                marginTop: 20
            }

export const MyAwesomeApp = () => {
    return (
        <div data-testid="div-app">
            <h1 data-testid="first-name-title">{firstName}</h1>
            <h3>{lastName}</h3>

            <p>{favoriteGames.join(', ')}</p>

            <h1>{isActive ? 'Active' : 'No active'}</h1>

            <p style={myStyles}>{JSON.stringify(address)}</p>
        </div>
    )
}