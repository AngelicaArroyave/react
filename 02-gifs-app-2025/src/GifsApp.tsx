import { useState } from 'react'

// import { mockGifs } from './mock-data/gifs.mock'
import { CustomHeader } from './shared/components/CustomHeader'
import { getGifsByQuery } from './gifs/actions/get-gifs-by-query.action'
import { GifList } from './gifs/components/GifList'
import { PreviousSearches } from './gifs/components/PreviousSearches'
import { SearchBar } from './shared/components/SearchBar'
import type { Gif } from './gifs/interfaces/gif.interface'

export const GifsApp = () => {
    const [previousTerms, setPreviousTerms] = useState<string[]>([])
    const [searchedGifs, setSearchedGifs] = useState<Gif[]>([])

    const handleTermClicked = (term: string) => {
        console.log("ðŸš€ ~ handleTermCliked ~ term:", term)
    }

    const handleSearch = async (query: string) => {
        query = query.toLowerCase().trim()

        if (!query) return

        if (previousTerms.includes(query)) return

        setPreviousTerms([query, ...previousTerms].slice(0, 8))

        const gifs = await getGifsByQuery(query)
        
        setSearchedGifs(gifs)
    }

    return (
        <>
            {/* Header */}
            <CustomHeader title='Gifs Search' description='Discover and share the perfect Gif' />

            {/* Search */}
            <SearchBar placeholder="Search for whatever you want" onQuery={handleSearch} />

            {/* Previous searches */}
            <PreviousSearches searches={previousTerms} onLabelClicked={handleTermClicked} />

            {/* Gifs */}
            <GifList gifs={searchedGifs} />
        </>
    )
}