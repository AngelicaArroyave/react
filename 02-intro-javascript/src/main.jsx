// Operador condicional ternario

const active = true

// Forma 1
const message = active ? 'Active' : 'Inactive'
console.log("ðŸš€ ~ message:", message)

// Forma 2
const message2 = !active && 'Active'
console.log("ðŸš€ ~ message2:", message2)

// Forma 3
const message3 = (active === true) && 'Active'
console.log("ðŸš€ ~ message3:", message3)