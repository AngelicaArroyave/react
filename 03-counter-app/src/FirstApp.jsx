import PropTypes from 'prop-types'

export const FirstApp = ({ title, subtitle, age, hobbie }) => {
    return (
        <>
            <h1> { title } </h1>
            <h2> { subtitle } </h2>
            <h2> { subtitle } </h2>
            <p> { age + 1 } </p>
            <p data-testid="test-hobbie"> { hobbie } </p>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    age: PropTypes.number.isRequired
}

// FirstApp.defaultProps = {
//     title: 'No title',
//     subtitle: 'No subtitle',
//     hobbie: 'Coding'
// }