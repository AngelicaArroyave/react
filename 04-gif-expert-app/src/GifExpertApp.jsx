import { AddCategory, GifGrid } from './components'
import { useState } from 'react'

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Isuku Midoriya'])

    const onAddCategory = (newCategory) => {
        if(categories.includes(newCategory)) return

        // Forma 1 con Concat
        // setCategories(categories.concat('Rick and Morty'))
        
        // Forma 2 con Spread Operator
        setCategories([newCategory, ...categories])
        // setCategories(categories => ['Rick and Morty', ...categories])
    }

    return (
        <>
            <h1> Gif Expert App </h1>

            <AddCategory onNewCategory={ value => onAddCategory(value) } />
            
            { categories.map(category => (
                <GifGrid key={ category } category={ category }/>
            )) }
        </>
    )
}