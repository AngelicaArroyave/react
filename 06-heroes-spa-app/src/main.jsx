import './app.css'
import { BrowserRouter } from 'react-router'
import React from 'react'
import ReactDOM from 'react-dom/client'

import { HeroesApp } from './HeroesApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <BrowserRouter>
      <HeroesApp />
    </BrowserRouter>
  </React.StrictMode>
)