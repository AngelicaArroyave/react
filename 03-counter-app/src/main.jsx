import './styles.css'
import { CounterApp } from './CounterApp'
import { FirstApp } from './FirstApp'
import React from 'react'
import ReactDOM from 'react-dom/client'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={10} />
        {/* <FirstApp title='Angie' age={25} hobbie='Coding'/> */}
    </React.StrictMode>
)