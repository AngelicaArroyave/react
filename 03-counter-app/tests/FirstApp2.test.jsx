import { FirstApp } from '../src/FirstApp'
import { render, screen } from '@testing-library/react'

describe('Test in component FirstApp', () => {
    
    const title = 'Angie'
    const subtitle = 'Hello!'
    const age = 25
    const hobbie = 'Coding'

    test('Should make macth with the snapshot', () => {
        const { container } = render(<FirstApp title={ title } age={ age } hobbie={ hobbie }/>)
        
        expect(container).toMatchSnapshot()
    })

    test('Should display the title', () => {
        render(<FirstApp title={ title } age={ age } hobbie={ hobbie }/>)
        // screen.debug()
        expect(screen.getByText(title)).toBeTruthy()
    })

    test('Should display the title as h1', () => {
        render(<FirstApp title={ title } age={ age } hobbie={ hobbie }/>)
        
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
    })

    test('Should display the title by ID', () => {
        render(<FirstApp title={ title } age={ age } hobbie={ hobbie }/>)

        expect(screen.getByText(hobbie)).toBeTruthy()
        expect(screen.getByTestId('test-hobbie').innerHTML).toContain(hobbie)
    })

    test('Should display the subtitle sent by props', () => {
        render(<FirstApp title={ title } subtitle={ subtitle } age={ age } hobbie={ hobbie }/>)

        expect(screen.getAllByText(subtitle).length).toBe(2)
    })
})