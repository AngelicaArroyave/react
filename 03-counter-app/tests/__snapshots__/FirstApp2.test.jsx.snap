// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component FirstApp Should make macth with the snapshot 1`] = `
<div>
  <h1>
     
    Angie
     
  </h1>
  <h2>
     
     
  </h2>
  <h2>
     
     
  </h2>
  <p>
     
    26
     
  </p>
  <p
    data-testid="test-hobbie"
  >
     
    Coding
     
  </p>
</div>
`;
